packages/in_app_purchase/android/src/main/java/io/flutter/plugins/inapppurchase/MethodChannelHandler.java

```diff
@@ -0,0 +1,262 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.plugins.inapppurchase;
+
+import static io.flutter.plugins.inapppurchase.Translator.fromPurchasesList;
+import static io.flutter.plugins.inapppurchase.Translator.fromPurchasesResult;
+import static io.flutter.plugins.inapppurchase.Translator.fromSkuDetailsList;
+
+import android.app.Activity;
+import android.content.Context;
+import android.util.Log;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import com.android.billingclient.api.BillingClient;
+import com.android.billingclient.api.BillingClientStateListener;
+import com.android.billingclient.api.BillingFlowParams;
+import com.android.billingclient.api.ConsumeResponseListener;
+import com.android.billingclient.api.Purchase;
+import com.android.billingclient.api.PurchaseHistoryResponseListener;
+import com.android.billingclient.api.SkuDetails;
+import com.android.billingclient.api.SkuDetailsParams;
+import com.android.billingclient.api.SkuDetailsResponseListener;
+import io.flutter.plugin.common.MethodCall;
+import io.flutter.plugin.common.MethodChannel;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/** Handles method channel for the plugin. */
+class MethodChannelHandler implements MethodChannel.MethodCallHandler {
```

Comment:
subjective: Not sure about the name here because it's not really handling method channels, it's more handling method calls for a given method channel. You could name it "MethodCallHandler" and it would work since this package is unique, but it may be confusing since there's another "MethodCallHandler" that it's implementing. I've used "MethodCallHandlerImpl" for this class name in other places but I'm not set on that being particularly great. There's also "InAppPurchaseMethodCallHandler"?